test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs

```diff
@@ -135,6 +197,81 @@ public void Constructor_InitializesStatusCodeAndValue(object value)
                 "No route matches the supplied values.");
         }
 
+        [Fact]
+        public void OnFormatting_NullUrlHelperContextHasRequestServices_ReturnsRequestServicesAction()
+        {
+            // Arrange
+            var context = new ActionContext(new DefaultHttpContext(), new RouteData(), new ActionDescriptor());
+            context.HttpContext.RequestServices = new ForwardingServiceProvider();
+
+            // Act
+            var result = new AcceptedAtActionResult(actionName: null, controllerName: null, routeValues: null, value: null);
+            result.OnFormatting(context);
+
+            // Assert
+            var header = context.HttpContext.Response.Headers.Last();
+            Assert.Equal("Location", header.Key);
+            Assert.Equal("abc", header.Value);
+        }
+
+        [Fact]
+        public void OnFormatting_NullUrlHelperContextNoRequestServices_ThrowsArgumentNullExeption()
+        {
+            // Arrange
+            var context = new ActionContext(new DefaultHttpContext(), new RouteData(), new ActionDescriptor());
+
+            // Act
+            var result = new AcceptedAtActionResult(actionName: null, controllerName: null, routeValues: null, value: null);
+
+            // Assert
+            Assert.Throws<ArgumentNullException>("provider", () => result.OnFormatting(context));
+        }
+
+        [Fact]
+        public void OnFormatting_NonNullUrlHelper_ReturnsUrlHelperAction()
```

Comment:
Ah yes - I got confused by all the uber complicated mocking code. I'll remove this test then